src/content/reference/react/useOptimistic.md

```diff
@@ -107,11 +108,15 @@ export default function App() {
   const [messages, setMessages] = useState([
     { text: "Hello there!", sending: false, key: 1 }
   ]);
-  async function sendMessage(formData) {
-    const sentMessage = await deliverMessage(formData.get("message"));
-    setMessages((messages) => [...messages, { text: sentMessage }]);
+  function sendMessageAction(formData) {
+    startTransition(async () => {
+      const sentMessage = await deliverMessage(formData.get("message"));
+      startTransition(() => {
+        setMessages((messages) => [{ text: sentMessage }, ...messages]);
+      })
+    })
   }
-  return <Thread messages={messages} sendMessage={sendMessage} />;
+  return <Thread messages={messages} sendMessageAction={sendMessageAction} />;
```

Comment:
This shouldn't really use the `Action` suffix convention if you do it this way because that implies that `Thread` wraps it in `startTransition` for you (which it could if you want to). That's what an Action is.

Reply:
https://github.com/reactjs/react.dev/pull/7796